### DISCUSSION TITLE
DNS Rewrite config issue /w ipv6?

### DISCUSSION THREAD

**Discussion by Dulanic** [AdguardTeam/AdGuardHome#3380] [Q&A]
Edit: This looks to impact AAAA only and the AGH DHCP doesn't have static IPs, I'm not sure what the workaround would be?

I am using adguard home for DNS and DHCP.  Since I get loopback issues /w websocks,, I put my server for DNS rewrites to the local IP. However, it is not working. If I add it to hosts on my windows box, its good, but nslookup is still pulling in cloudflare IP instead. It looks to be the DHCP part that is not working. The logs show the DNS is rewritten as expected, but windows still comes back with the cloudflare IP. What is throwing me off is Adguard is the DHCP so it sets the DNS provider and ipconfig looks good as far as I can tell.

If I run it in WSL2, it pulls in the correct info, so it seems to be something /w windows but that may be because WSL2 uses ipv4? Since this doesn't seem to be a adguard issue so much as a configuration issue, I opened here instead of issue. It is coming back /w a DNS server /w a ipv6 address which is actually my own PC.... why is this happening?

![image](https://user-images.githubusercontent.com/2948206/127022238-7be26130-8f35-4f6e-9770-c2b263d86f81.png)

![image](https://user-images.githubusercontent.com/2948206/127021088-d6a1a358-fd23-4319-b6fa-4520af01c16d.png)

![image](https://user-images.githubusercontent.com/2948206/127020292-d4845c55-09a2-4aff-bfaa-81c141f4f817.png)

![image](https://user-images.githubusercontent.com/2948206/127020444-775974ce-8e79-45a1-8be3-ae873f8362b0.png)

![image](https://user-images.githubusercontent.com/2948206/127019722-0bf69dbb-05ac-4fa2-8fa3-ffa15ff7e00c.png)

**Comment by ainar-g** [comment]
Hello and thank you for the report.  Which version of AdGuard Home are you using?  v0.107.0-b.5 was shipped with some bug fixes specifically with regards to DNS rewrites and IPv6.

**Reply by Dulanic** [reply]
I am running v0.107.0-b.5. It looks like it is replying /w ipv6 cloudflare address most the time. I can work around this by turning off ipv6 reponses, but shouldnt rewrites take priority over that?

**Reply by ainar-g** [reply]
Hello again and apologies for the long time without any updates.  After carefully analysing the situation, we have a theory that this has something to do with ongoing issues with how we currently process AAAA requests for DHCP hosts.  Do you use the address of AGH as **both** the primary **and** the secondary DNS server?  It might be that Windows receives no answer from AGH and tries to perform the same request to the secondary DNS server, receiving the Cloudflare ones.
